tests/integration/test_charm.py

```diff
@@ -240,121 +224,45 @@ async def test_relations(
     assert: It should have the correct status
     """
 
-    def test_discourse_srv_status_ok():
-        response = requests.get(f"{discourse_address}/srv/status", timeout=requests_timeout)
-        assert response.status_code == 200
+    def srv_status():
+        return requests.get(f"{discourse_address}/srv/status", timeout=requests_timeout)
+
+    def srv_status_raises_connection_error():
+        try:
+            srv_status()
+            return False
+        except requests.ConnectionError:
+            return True
 
     # The charm should be active when starting this test
-    await model.wait_for_idle(status="active")
-    test_discourse_srv_status_ok()
+    juju.wait(jubilant.all_active)
+    assert srv_status().status_code == 200
 
     # Removing the relation to postgresql should disable the charm
-    await model.applications[app.name].remove_relation("database", "postgresql-k8s:database")
-    await model.wait_for_idle(apps=[app.name], status="waiting")
-    with pytest.raises(requests.ConnectionError):
-        test_discourse_srv_status_ok()
+    juju.remove_relation(app.name, "postgresql-k8s:database")
+    juju.wait(
+        lambda status: status.apps[app.name].is_waiting and srv_status_raises_connection_error()
```

Comment:
Given that this condition is used twice, what do you feel about something like:

```
juju.wait(not_ready)

def not_ready(status: jubilant.Status) -> bool:
    if not status.apps[app_name].is_waiting:
        return False
    try: ...
    except (...): return True
```

Reply:
Personally I think that's less clear than what I have, but happy for the discourse-k8s team to chime in.